### ISSUE TITLE
OPNsense deploy to ProxmoxVE Fails

### ISSUE THREAD

**Issue by theoisadoor** [acmesh-official/acme.sh#4983]
Deploying a certificate (whether issuing new, or starting automation for existing certificate) from OPNsense to ProxmoxVE fails with PHP fatal error. Have tested with new user 'acme' and existing root account, setting permissions global ('/') or restricted to nodes ('/nodes') on ProxmoxVE, settings permissions at token level or user level. Always results in same error:

```
[02-Feb-2024 16:32:08 Europe/London] PHP Fatal error:  Uncaught TypeError: proc_open(): Argument #2 ($descriptor_spec) must be of type array, null given in /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php:133
Stack trace:
#0 /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php(133): proc_open('/usr/local/sbin...', NULL, NULL, NULL, Array)
#1 /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php(106): OPNsense\AcmeClient\LeAutomation\Base->runAcme()
#2 /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeCertificate.php(606): OPNsense\AcmeClient\LeAutomation\Base->run()
#3 /usr/local/opnsense/scripts/OPNsense/AcmeClient/lecert.php(188): OPNsense\AcmeClient\LeCertificate->runAutomations()
#4 /usr/local/opnsense/scripts/OPNsense/AcmeClient/lecert.php(199): main()
#5 {main}
  thrown in /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php on line 133
```
I have been verifying config using the following guide:
https://github.com/acmesh-official/acme.sh/wiki/deployhooks#27-deploy-the-cert-on-a-proxmox-ve-node

Setup Information
------------------
```
Kernel Version Linux 5.15.136-1-pve #1 SMP PVE 5.15.136-1 (2024-01-23T11:10Z)
PVE Manager Version pve-manager/7.4-17/513c62be
```
_No firewall rules configured on ProxmoxVE to prevent OPNsense from communicating with the admin interface._

```
User-Agent Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
FreeBSD 13.2-RELEASE-p9 stable/24.1-n254969-8659880248c SMP amd64
OPNsense 24.1_1 2dba37306
Plugins os-acme-client-4.0 os-apcupsd-1.1 os-clamav-1.8 os-crowdsec-1.0.7 os-dmidecode-1.1_1 os-iperf-1.0_1 os-node_exporter-1.2 os-nut-1.8.1_2 os-theme-cicada-1.34_4
Time Fri, 02 Feb 2024 16:32:15 +0000
OpenSSL 3.0.12
Python 3.9.18
PHP 8.2.15
```

Debug level has been set to `debug 3` in OPNsense ACME client settings, but I cannot find where the logs are output. Any help pointing me in the right direction so I can post them would be greatly appreciate.

Steps to reproduce
------------------

1. Create API token for root in ProxmoxVE, do not enable Priviledge Seperation
![Screenshot 2024-02-02 161844](https://github.com/acmesh-official/acme.sh/assets/156682752/fcd8bdb2-24db-425b-aaa6-0f0f04e2ce62)
2. Add token permission for root with PVECertificate role (sys.modify)
![Screenshot 2024-02-02 161731](https://github.com/acmesh-official/acme.sh/assets/156682752/980662b6-2579-44a0-be7b-47605192d328)
3. Configure Automation on OPNsense
![Screenshot 2024-02-02 162452](https://github.com/acmesh-official/acme.sh/assets/156682752/91e3266a-5468-4955-8258-64fe6e7d61e5)
4. Attach automation to certificate profile in OPNsense
5. 'Run automations' on certificate in OPNsense
6. OPNsense 'System Status' reports error
![Screenshot 2024-02-02 163810](https://github.com/acmesh-official/acme.sh/assets/156682752/c093aa03-2ded-43dc-9e7d-732aa94e024c)

Debug log
-----------------
I have attempted to run the command from the OPNsense shell but it is declaring that 'The deploy hook proxmoxve is not found.'. This deployment script is what is reported from the ACME 'System Log' file on OPNsense, so unsure if this will report correctly.
![Screenshot 2024-02-02 165158](https://github.com/acmesh-official/acme.sh/assets/156682752/7cf17d49-f085-4b90-846b-657aea031544)
```
acme.sh --deploy --syslog 7 --debug 3 --server 'letsencrypt' --home '/var/etc/acme-client/home' --cert-home '/var/etc/acme-client/cert-home/6505a18a50ef89.49267983' --certpath '/var/etc/acme-client/certs/6505a18a50ef89.49267983/cert.pem' --keypath '/var/etc/acme-client/keys/6505a18a50ef89.49267983/private.key' --capath '/var/etc/acme-client/certs/6505a18a50ef89.49267983/chain.pem' --fullchainpath '/var/etc/acme-client/certs/6505a18a50ef89.49267983/fullchain.pem' --domain 'HL1-MGT-PVRT11.domain.here' --deploy-hook proxmoxve
[Fri Feb  2 16:46:17 GMT 2024] _is_idn_d='HL1-MGT-PVRT11.domain.here'
[Fri Feb  2 16:46:17 GMT 2024] _idn_temp
[Fri Feb  2 16:46:17 GMT 2024] _selectServer try snames='zerossl.com,zerossl'
[Fri Feb  2 16:46:17 GMT 2024] _selectServer try snames='letsencrypt.org,letsencrypt'
[Fri Feb  2 16:46:17 GMT 2024] _selectServer match letsencrypt
[Fri Feb  2 16:46:17 GMT 2024] Selected server: https://acme-v02.api.letsencrypt.org/directory
[Fri Feb  2 16:46:17 GMT 2024] readlink exists=0
[Fri Feb  2 16:46:17 GMT 2024] dirname exists=0
[Fri Feb  2 16:46:17 GMT 2024] Lets find script dir.
[Fri Feb  2 16:46:17 GMT 2024] _SCRIPT_='/usr/local/sbin/acme.sh'
[Fri Feb  2 16:46:17 GMT 2024] _script='/usr/local/sbin/acme.sh'
[Fri Feb  2 16:46:17 GMT 2024] _script_home='/usr/local/sbin'
[Fri Feb  2 16:46:17 GMT 2024] Using config home:/var/etc/acme-client/home
[Fri Feb  2 16:46:17 GMT 2024] ACCOUNT_CONF_PATH='/var/etc/acme-client/home/account.conf'
[Fri Feb  2 16:46:17 GMT 2024] logger exists=0
[Fri Feb  2 16:46:17 GMT 2024] OK
[Fri Feb  2 16:46:17 GMT 2024] 1:SYS_LOG='7'
[Fri Feb  2 16:46:17 GMT 2024] LE_WORKING_DIR='/var/etc/acme-client/home'
https://github.com/acmesh-official/acme.sh
v3.0.7
[Fri Feb  2 16:46:17 GMT 2024] Using server: https://acme-v02.api.letsencrypt.org/directory
[Fri Feb  2 16:46:17 GMT 2024] Running cmd: deploy
[Fri Feb  2 16:46:17 GMT 2024] Using config home:/var/etc/acme-client/home
[Fri Feb  2 16:46:17 GMT 2024] ACCOUNT_CONF_PATH='/var/etc/acme-client/home/account.conf'
[Fri Feb  2 16:46:17 GMT 2024] ACME_DIRECTORY='https://acme-v02.api.letsencrypt.org/directory'
[Fri Feb  2 16:46:17 GMT 2024] _ACME_SERVER_HOST='acme-v02.api.letsencrypt.org'
[Fri Feb  2 16:46:17 GMT 2024] _ACME_SERVER_PATH='directory'
[Fri Feb  2 16:46:17 GMT 2024] CA_CONF='/var/etc/acme-client/home/ca/acme-v02.api.letsencrypt.org/directory/ca.conf'
[Fri Feb  2 16:46:17 GMT 2024] DOMAIN_PATH='/var/etc/acme-client/cert-home/6505a18a50ef89.49267983/HL1-MGT-PVRT11.domain.here'
[Fri Feb  2 16:46:17 GMT 2024] DOMAIN_CONF='/var/etc/acme-client/cert-home/6505a18a50ef89.49267983/HL1-MGT-PVRT11.domain.here/HL1-MGT-PVRT11.domain.here.conf'
[Fri Feb  2 16:46:17 GMT 2024] OK
[Fri Feb  2 16:46:17 GMT 2024] 34:Le_DeployHook='proxmoxve,'
[Fri Feb  2 16:46:17 GMT 2024] The deploy hook proxmoxve is not found.
```

**Comment by theoisadoor** [comment]
> Please upgrade to the latest code and try again first. Maybe it's already fixed. `acme.sh --upgrade` If it's still not working, please provide the log with `--debug 2`, otherwise, nobody can help you.

`acme.sh --upgrade` was run, no change. `--debug 3` was already specified, which according to docs is more verbose.

**Comment by Qhilm** [comment]
Maybe unrelated, but I have the same crash error message when attempting to upload a cert to an AVM Fritz!Box. Interestingly, this was working fine a couple of months ago, on exactly the same opnsense box. Unfortunately I don't know which version of acme.sh was running on the opnsense at this time. opnsense was also upgraded a couple of times since then, currently on 24.1_1

At the moment, it's running acme.sh 3.0.7.


>`[06-Feb-2024 09:00:10 Europe/Berlin] PHP Fatal error:  Uncaught TypeError: proc_open(): Argument #2 ($descriptor_spec) must be of type array, null given in /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php:133`
>`Stack trace:`
>`#0 /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php(133): proc_open('/usr/local/sbin...', NULL, NULL, NULL, Array)`
>`#1 /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php(106): OPNsense\AcmeClient\LeAutomation\Base->runAcme()`
>`#2 /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeCertificate.php(606): OPNsense\AcmeClient\LeAutomation\Base->run()`
>`#3 /usr/local/opnsense/scripts/OPNsense/AcmeClient/lecert.php(188): OPNsense\AcmeClient\LeCertificate->runAutomations()`
>`#4 /usr/local/opnsense/scripts/OPNsense/AcmeClient/lecert.php(199): main()`
>`#5 {main}`
>`  thrown in /usr/local/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/Base.php on line 133`

**Comment by towerhand** [comment]
This issue will be fixed in the upcoming opnsense acme-client 4.1.

https://github.com/opnsense/plugins/issues/3790
